### PR TITLE
Publish multivalue readings as string instead of an array

### PR DESCRIPTION
**Pull request by milanzlicic**

### CODE REVIEW

File: src/test/java/com/wolkabout/wolk/protocol/WolkaboutProtocolTest.java [hunk 1]
```diff
@@ -26,15 +27,50 @@
...
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import static org.mockito.Mockito.*;
 
+class MessageMatcher implements ArgumentMatcher<byte[]> {
+
+    String regex;
+    String arg;
```
Comment (nanavuletic) [inline src/test/java/com/wolkabout/wolk/protocol/WolkaboutProtocolTest.java:46]: Is `arg` not necessary here?

File: src/test/java/com/wolkabout/wolk/protocol/WolkaboutProtocolTest.java [hunk 2]
```diff
@@ -26,15 +27,50 @@
...
+
+    public MessageMatcher(String regex) {
+        this.regex = regex;
```
Comment (nanavuletic) [inline src/test/java/com/wolkabout/wolk/protocol/WolkaboutProtocolTest.java:49]: And also, couldn't you store `Pattern` as a member? Not make `Pattern.compile` on every instance?

**Review by nanavuletic — APPROVED** [review]
Nice to have changes, optimizations...

File: src/test/java/com/wolkabout/wolk/protocol/WolkaboutProtocolTest.java (hunk 1 above)
Comment (milanzlicic) [reply src/test/java/com/wolkabout/wolk/protocol/WolkaboutProtocolTest.java:46]: We store when match is called so we can output it in toString method to have a more meaningful message if match fails

**Pull request merged**